lib/util/util.c

```diff
@@ -723,11 +723,12 @@ int prog_lock_get(const char *progname)
 				return err;
 			}
 
-			len = read(fd, buf, sizeof(buf));
+			len = read(fd, buf, sizeof(buf)-1);
 			close(fd);
-			buf[len] = '\0';
-
-			pid = strtoul(buf, NULL, 10);
+			if (len > 0) {
+				buf[len] = '\0';
+				pid = strtoul(buf, NULL, 10);
+			}
 			if (!pid) {
```

Comment:
Bit of a nity comment, but why not check errno here also, as ret on error could be ULONG_MAX.
So `if (!pid || errno)`

Reply:
ACK, can do